driver/docker/docker.go

```diff
@@ -50,10 +51,36 @@ func (d *Driver) AddConfigurationOptions(opts ...ConfigurationOption) {
 	d.dockerConfigurationOptions = append(d.dockerConfigurationOptions, opts...)
 }
 
-// GetContainerConfig returns the container configuration and host configuration
+// GetContainerConfig returns a copy of the container configuration
 // used by the driver during container exec
-func (d *Driver) GetContainerConfig() (*container.Config, *container.HostConfig) {
-	return d.containerCfg, d.containerHostCfg
+func (d *Driver) GetContainerConfig() (container.Config, error) {
+	cpy, err := copystructure.Copy(*d.containerCfg)
+	if err != nil {
+		return container.Config{}, err
+	}
+
+	containerCfg, ok := cpy.(container.Config)
+	if !ok {
+		return container.Config{}, err
```

Comment:
Here you are returning `err` but it will be `nil` but the returned config doesn't include the configuration data requested which is an error. Shouldn't this return an `errors.New` instead? Same for the method below.

Reply:
Indeed, thank you for catching.  Updated.